File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
from heapq import heappush, heappop, heapify

def solution(scoville, K):
    answer = 0
    heapify(scoville)
    while scoville[0] < K:
        lowest = heappop(scoville)
        second_lowest = heappop(scoville)
        new_spicy = lowest + second_lowest * 2
        heappush(scoville, new_spicy)
        answer += 1
        if len(scoville) == 1 and scoville[0] < K:
            answer = -1
            break
            
    return answer

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;
import java.util.Arrays;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < scoville.length; i++) {
            pq.add(scoville[i]);
        }
        
        while (pq.peek() < K) {
            int lowest = pq.poll();
            int secondLowest = pq.poll();
            int newSpicy = lowest + secondLowest * 2;
            pq.add(newSpicy);
            answer += 1; 
            if (pq.size() == 1 && pq.peek() < K) {
                answer = -1;
                break;
            }
        }
        return answer;
    }
}